server/src/main/java/org/elasticsearch/search/sort/FieldSortBuilder.java

```diff
@@ -332,6 +332,9 @@ public SortFieldAndFormat build(QueryShardContext context) throws IOException {
 
             final Nested nested;
             if (nestedSort != null) {
+                if (context.indexVersionCreated().before(Version.V_6_5_0) && nestedSort.getMaxChildren() != Integer.MAX_VALUE) {
```

Comment:
I just realized that we should also limit this new option to the last level of nested sort. So if `nestedSort.getNestedSort()` is not null and `max_children` is set we should throw an error.

Reply:
Hi @jimczi ,
I just curious about that limitation. Why we do that?
Cause when i look at the code, child nested's use for filtering data. They cannot use max_children property. Maybe we need to set max_children just on top level parent nested and throwing error when set child nested.

Am i right?